Source/IntegrationTests/TestFiles/LitTests/LitTest/comp/rust/conversions.dfy

```diff
@@ -27,10 +29,12 @@ method TestCompile(input: Option<Uint8>) returns (output: Option<Uint8>) {
   return None;
 }
 
+type CustomBv16 = bv16
+
 method Main() {
   var x: Uint8 := 2;
   // Correct rendering of expressions that can interpret the first '<' as a generic argument
-  var y: bv16 := (x as bv16) << 2;
+  var y: bv16 := (x as CustomBv16) << 2;
```

Comment:
Is this working around another bug?

Reply:
Excellent question. Thanks for asking.
Looking at the generated code, CustomBv16 is actually rendered as a name as well, and the Dafny-to-Rust correctly generates parentheses around the as expression because otherwise the "<" would be interpreted as the start of a generic argument.
To sum up, I'm still checking the other (already fixed) bug, just adding another kind of type to verify types, which reproduces the same issue I had originally found when compiling this line to Rust:

https://github.com/dafny-lang/dafny/blob/ad994713edde359976bd1e813718fc5292a60e3b/Source/DafnyStandardLibraries/src/Std/Unicode/Utf8EncodingForm.dfy#L180